src/compiler/scanner.ts

```diff
@@ -27,7 +27,8 @@ namespace ts {
         getTokenFlags(): TokenFlags;
         reScanGreaterToken(): SyntaxKind;
         reScanSlashToken(): SyntaxKind;
-        reScanTemplateToken(): SyntaxKind;
+        reScanTemplateToken(isTaggedTemplate?: boolean): SyntaxKind;
+        reScanTemplateHead(): SyntaxKind;

```

Comment:
@mhegazy, thoughts on just having `reScanTemplateToken` operate on template heads as well? The implementation could have a `Debug.assert` like so: